src/components/Api.js

```diff
@@ -0,0 +1,94 @@
+export default class Api {
+  //---------------------------API CONSTRUCTOR------------------------------->>
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  //-----------------METHOD TO CHECK RESPONSE FROM SERVER------------------>>
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error :${res.status}`);
+  }
+
+  //-----------------METHOD TO GET USER INFO------------------>>
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then((res) => {
+      return this._checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.